TrueCar/gluestick
# add bash completion

It's kind-of working:
![gluestick_bash_completion](https://user-images.githubusercontent.com/1627760/30001718-6c10c872-905b-11e7-86ea-de77c5ecf8b4.gif)

## testing
The unit tests cover the javascript only; they mimic the input format of the bash function.  To functional test the completion against a full project (with multiple entry points), do the following: 

- Check out this branch and override the globally-installed gluestick-cli.
- [Source the completion script and override the location variable.](https://github.com/TrueCar/gluestick/pull/1097/files#diff-733de6a3eeea98143a4fe25bf6f76611R5)
- Update the project's gluestick files, (see [node_modules/gluestick/build/cli.js](https://github.com/TrueCar/gluestick/pull/1097/files#diff-6a60c385a9090ce5fe3dfb036a180ac4) and [this upgrade warning](https://github.com/TrueCar/gluestick/pull/1097/files#diff-497b0fdf7b6939feac1cac1572f6312bR66) )

## The plan so far:

1. Create a bash script to be used as the bash completion entry
2. Call a node script with the full set of completion words
3. Reflect on the commander configuration for global and local gluestick to return completion

## Installation

in `~/.bashrc` or `~/.zsh.rc` or whichever:
`source $(dirname $(readlink -f $(which gluestick)))/completion.sh`
(beware of freebsd problems with readlink)

or, from `/etc/bash_completion.d/`:
`ln -s /usr/bin/node_modules/gluestick/bin/completion.sh gluestick`
